src/Microsoft.ML.TimeSeries/PredictionEngine.cs

```diff
@@ -0,0 +1,207 @@
+﻿using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.ML.TimeSeries
+{
+    public delegate void ValuePinger(ref bool value, int rowPosition);
+
+    public interface IStatefulRow : IRow
+    {
+        ValuePinger[] GetPingers();
+    }
+
+    public interface IStatefulRowMapper : IRowMapper
+    {
+        Delegate[] CreatePingers(IRow input, Func<int, bool> activeOutput, out Action disposer);
```

Comment:
>Delegate[] CreatePingers(IRow input, Func<int, bool> activeOutput, out Action disposer); [](start = 8, length = 88)

why do you need to have this method? I think, regardless of anything, you must call the pinger